src/actions/authActions.js

```diff
@@ -102,6 +102,16 @@ export const updateFcmTokenAction = (walletId: string) => {
   };
 };
 
+export const proceedWithPkAction = (pin: string, action: (pk: string) => void) => {
+  return async (dispatch: Dispatch) => {
+    const { wallet: encryptedWallet } = await storage.get('wallet');
+    const saltedPin = await getSaltedPin(pin, dispatch);
+    const wallet = await decryptWallet(encryptedWallet, saltedPin);
+    const pk = get(wallet, 'signingKey.privateKey');
+    action(pk);
+  };
+};
```

Comment:
Shouldn't this retrieve not from storage, but from keychain (if it's not saved with biometric flag)?

Reply:
It's the only place we know for sure that wallet exists, so I assumed it's the safest way to go.
And faster than waiting it to be fetched from keychain. No?